### DISCUSSION TITLE
Invalid regex expression

### DISCUSSION THREAD

**Discussion by tan-wen** [apache/doris#7894] [RFC]
doris 版本 【select verison() 】 5.1.0

字段使用 not regexp 匹配时，返回
`> 1064 - errCode = 2, detailMessage = Invalid regex expression^(?:(?!0000)[0-9]{4}-(?:(?:0[1-9]|1[0-2])-(?:0[1-9]|1[0-9]|2[0-8])|(?:0[13-9]|1[0-2])-(?:29|30)|(?:0[13578]|1[02])-31)|(?:[0-9]{2}(?:0[48]|[2468][048]|[13579][26])|(?:0[48]|[2468][048]|[13579][26])00)-02-29)$`

该正则表达式在 MySQL 8.0.25 可以正确解析，匹配结果。
请问，Doris在正则表达式上有特殊什么限制吗？

**Comment by tan-wen** [comment]
使用的正则表达式是进行日期格式校验的，yyyy-MM-dd

`^(?:(?!0000)[0-9]{4}-(?:(?:0[1-9]|1[0-2])-(?:0[1-9]|1[0-9]|2[0-8])|(?:0[13-9]|1[0-2])-(?:29|30)|(?:0[13578]|1[02])-31)|(?:[0-9]{2}(?:0[48]|[2468][048]|[13579][26])|(?:0[48]|[2468][048]|[13579][26])00)-02-29)$`

**Comment by BiteTheDDDDt** [comment]
maybe the problem is  `(?!0000)`.
doris use re2 to process regex now, you can read this issue.  https://github.com/google/re2/issues/288

**Reply by tan-wen** [reply]
thanks
